src/components/table/JadeTable.jsx

```diff
@@ -4,98 +4,184 @@ import { withStyles } from '@material-ui/core/styles';
+    paddingLeft: theme.spacing.unit * 3,
+  },
+  searchInput: {
+    paddingTop: theme.spacing.unit * 1.5,
+    paddingLeft: theme.spacing.unit * 6,
   },
 });
 
-const desc = (a, b, orderBy) => {
-  if (b[orderBy] < a[orderBy]) {
-    return -1;
-  }
-  if (b[orderBy] > a[orderBy]) {
-    return 1;
-  }
-  return 0;
-};
-
-const stableSort = (array, cmp) => {
-  const stabilizedThis = array.map((el, index) => [el, index]);
-  stabilizedThis.sort((a, b) => {
-    const order = cmp(a[0], b[0]);
-    if (order !== 0) return order;
-    return a[1] - b[1];
-  });
-  return stabilizedThis.map(el => el[0]);
-};
-
-const getSorting = (order, orderBy) =>
-  order === 'desc' ? (a, b) => desc(a, b, orderBy) : (a, b) => -desc(a, b, orderBy);
-
 export class JadeTable extends React.PureComponent {
   state = {
-    order: 'asc',
-    orderBy: 'lastModified',
+    orderDirection: '',
+    orderBy: '',
+    page: 0,
+    rowsPerPage: 5,
+    searchString: '',
   };
 
   static propTypes = {
     classes: PropTypes.object.isRequired,
     columns: PropTypes.arrayOf(PropTypes.object),
+    handleFilter: PropTypes.func,
     rows: PropTypes.arrayOf(PropTypes.object),
+    summary: PropTypes.bool,
+    totalCount: PropTypes.number.isRequired,
   };
 
-  handleRequestSort = (event, property) => {
-    const { order, orderBy } = this.state;
+  handleRequestSort = (event, sort) => {
+    const { handleFilter } = this.props;
+    const { orderDirection, orderBy, page, rowsPerPage, searchString } = this.state;
     let newOrder = 'desc';
-    if (orderBy === property && order === 'desc') {
+    if (orderBy === sort && orderDirection === 'desc') {
       newOrder = 'asc';
     }
+    const offset = page * rowsPerPage;
+    this.setState({ orderDirection: newOrder, orderBy: sort });
+    handleFilter(rowsPerPage, offset, sort, newOrder, searchString);
+  };
+
+  handleChangeRowsPerPage = event => {
+    const { handleFilter } = this.props;
+    const { orderDirection, orderBy, page, searchString } = this.state;
+    const limit = event.target.value;
+    const offset = page * limit;
+    this.setState({ rowsPerPage: limit });
+    handleFilter(limit, offset, orderBy, orderDirection, searchString);
+  };
+
+  handleChangePage = (event, page) => {
+    const { handleFilter } = this.props;
+    const { orderDirection, orderBy, rowsPerPage, searchString } = this.state;
+    const offset = page * rowsPerPage;
+    this.setState({ page }); // offset
+    handleFilter(rowsPerPage, offset, orderBy, orderDirection, searchString);
+  };
 
-    this.setState({ order: newOrder, orderBy: property });
+  handleSearchString = event => {
+    const { handleFilter } = this.props;
+    const { page, orderDirection, orderBy, rowsPerPage } = this.state; // limit
+    const offset = page * rowsPerPage;
+    const searchString = event.target.value;
+    this.setState({ searchString }); // filter
+    handleFilter(rowsPerPage, offset, orderBy, orderDirection, searchString);
```

Comment:
This is not specific to just filtering -- let's call it something like `handleEnumeration`?